satpy/tests/compositor_tests/test_viirs.py

```diff
@@ -223,6 +242,254 @@ def test_hncc_dnb(self):
         np.testing.assert_allclose(
             unique, [3.484797e-04, 9.507845e-03, 4.500016e+03])
 
+    def test_reflectance_corrector_abi(self):
+        import xarray as xr
+        import dask.array as da
+        import numpy as np
+        from satpy.composites.viirs import ReflectanceCorrector
+        from satpy import DatasetID
+        ref_cor = ReflectanceCorrector(dem_filename='_fake.hdf', optional_prerequisites=[
+            DatasetID(name='satellite_azimuth_angle', wavelength=None, resolution=None, polarization=None,
+                      calibration=None,
+                      level=None, modifiers=None),
+            DatasetID(name='satellite_zenith_angle', wavelength=None, resolution=None, polarization=None,
+                      calibration=None,
+                      level=None, modifiers=None),
+            DatasetID(name='solar_azimuth_angle', wavelength=None, resolution=None, polarization=None, calibration=None,
+                      level=None, modifiers=None),
+            DatasetID(name='solar_zenith_angle', wavelength=None, resolution=None, polarization=None, calibration=None,
+                      level=None, modifiers=None)], name='C01', prerequisites=[], wavelength=(0.45, 0.47, 0.49),
+                                       resolution=1000, calibration='reflectance',
+                                       modifiers=('sunz_corrected', 'rayleigh_corrected_crefl',), sensor='abi')
+
+        self.assertEqual(ref_cor.attrs['modifiers'], ('sunz_corrected', 'rayleigh_corrected_crefl',))
+        self.assertEqual(ref_cor.attrs['calibration'], 'reflectance')
+        self.assertEqual(ref_cor.attrs['wavelength'], (0.45, 0.47, 0.49))
+        self.assertEqual(ref_cor.attrs['name'], 'C01')
+        self.assertEqual(ref_cor.attrs['resolution'], 1000)
+        self.assertEqual(ref_cor.attrs['sensor'], 'abi')
+        self.assertEqual(ref_cor.attrs['prerequisites'], [])
+        self.assertEqual(ref_cor.attrs['optional_prerequisites'], [
+            DatasetID(name='satellite_azimuth_angle', wavelength=None, resolution=None, polarization=None,
```

Comment:
The default for DatasetID is `None` so these can all be left empty (except for maybe `modifiers` which defaults to an empty tuple).